src/browser/provider/built-in/base.js

```diff
@@ -0,0 +1,69 @@
+import browserNatives from 'testcafe-browser-natives';
+import OS from 'os-family';
+import WARNING_MESSAGE from '../../../warnings/message';
+
+
+/*eslint-disable no-undef*/
+function getDataFromBrowser () {
+    return { width: window.innerWidth, height: window.innerHeight, title: document.title };
+}
+/*eslint-disable no-undef*/
+
+const INIT_SCRIPT = getDataFromBrowser.toString();
+
+
+export default class BrowserProviderBase {
+    constructor () {
+        // HACK: Browser window has different border size in normal and maximized modes. So we need to be sure that window is
+        // not maximized before resizing in order to keep client area size correction mechanism working. When browser is started,
+        // we are resizing it first time to switch the window to normal mode, and second - to restore the client area size.
+        this.resizeCorrections = {};
+    }
+
+    async calculateResizeCorrections (id) {
+        await this.waitForConnectionReady(id);
+
+        var { width, height, title } = await this.runInitScript(id, INIT_SCRIPT);
+
+        if (!await browserNatives.isMaximized(title))
+            return;
+
+        await browserNatives.resize(title, width, height, width, height);
+
+        var { width: newWidth, height: newHeight } = await this.runInitScript(id, INIT_SCRIPT);
+
+        await browserNatives.maximize(title);
+
+        if (newWidth === width && newHeight === height)
+            return;
+
+        this.resizeCorrections[id] = { width: newWidth - width, height: newHeight - height };
+    }
+
+    async resizeWindow (id, pageInfo, width, height) {
+        // TODO: remove once https://github.com/DevExpress/testcafe-browser-natives/issues/12 implemented
+        if (OS.linux) {
+            this.reportWarning(id, WARNING_MESSAGE.browserManipulationsNotSupportedOnLinux);
+            return;
+        }
+
+        if (this.resizeCorrections[id]) {
+            width -= this.resizeCorrections[id].width;
+            height -= this.resizeCorrections[id].height;
+
+            delete this.resizeCorrections[id];
+        }
+
+        await browserNatives.resize(pageInfo.title, pageInfo.width, pageInfo.height, width, height);
+    }
+
+    async screenshotWindow (id, pageInfo, screenshotPath) {
```

Comment:
Andrey said that "browser" should configure in the method name as in the openBrowser and closeBrowser methods. I'd use "takeScreenshot",  we've used such a method in browser-natives as well.